db/migrate/20180216204115_create_votes.rb

```diff
@@ -0,0 +1,10 @@
+class CreateVotes < ActiveRecord::Migration[5.1]
+  def change
+    create_table :votes do |t|
+      t.references :voteable, polymorphic: true, index: true
+      t.references :user, null: false
+      t.integer :value, null: false
+      t.timestamps
+    end
```

Comment:
might be good to add a uniqueness constraint here as well.

```ruby
create_index :votes, [:user_id, :voteable_id, :voteable_type], unique: true
```

remove the `index: true` on the `voteable` references if you end up going that route.